pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -572,20 +587,32 @@ func (plugin *kubenetNetworkPlugin) checkCNIPluginInDir(dir string) bool {
 	return true
 }
 
-// Returns a list of pods running or ready to run on this node and each pod's IP address.
-// Assumes PodSpecs retrieved from the runtime include the name and ID of containers in
-// each pod.
-func (plugin *kubenetNetworkPlugin) getActivePods() ([]*hostport.ActivePod, error) {
+// getNonExitedPods returns a list of pods running or ready to run on this node
```

Comment:
What "ready to run" means here is unclear...How about just "returns a list of pods where there are at least one running container in the pod"? 

On the other hand, as I said in line 679, you probably just want a list of "non-terminated" infra containers. You can filter out pods with a terminated or non-existent infra container in one place.